### PR TITLE
Feature/cargo

### PR DESCRIPTION
**Pull request by crowelch**

### CODE REVIEW

File: test/tests.js [hunk 1]
```diff
@@ -802,4 +802,48 @@ describe('pasync', function() {
...
+		};
+	});
+
+	it('cargo with error', function(done) {
+		var responseArray = [];
+		var cargo = pasync.cargo(function(tasks) {
+			return new Promise(function(resolve, reject) {
+				tasks.forEach(function(task) {
+					if(task === 3) {
+						reject(123);
```
Comment (crispy1989) [inline test/tests.js]: This should not be inside the inner forEach loop.  Same with the resolve below.

**Comment by crowelch** [comment]
I fixed it!

File: test/tests.js [hunk 2]
```diff
@@ -802,4 +802,53 @@ describe('pasync', function() {
...
+		cargo.push([1, 2, 3, 4, 5]).catch(done);
+		cargo.drain = function() {
+			expect(responseArray).to.deep.equal([12, 24, 36, 48, 60]);
+			done();
+		};
+	});
+
+	it('cargo with error', function(done) {
+		var responseArray = [];
+		var isError = false;
```
Comment (crispy1989) [inline test/tests.js]: Declare this in the innermost scope it's used in?

**Comment by crowelch** [comment]
Fixed and fixed

**Pull request merged**